eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/models/BomDependencyNoVersion.java

```diff
@@ -0,0 +1,93 @@
+package com.azure.tools.bomgenerator.models;
+
+import org.jboss.shrinkwrap.resolver.api.maven.PackagingType;
+import org.jboss.shrinkwrap.resolver.api.maven.ScopeType;
+import org.jboss.shrinkwrap.resolver.api.maven.coordinate.MavenCoordinate;
+import org.jboss.shrinkwrap.resolver.api.maven.coordinate.MavenDependency;
+import org.jboss.shrinkwrap.resolver.api.maven.coordinate.MavenDependencyExclusion;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class BomDependencyNoVersion implements MavenDependency {
```

Comment:
Instead of making a sub-type of `MavenDependency`, could we track information differently to support the additional values we want to add into the class?

Reply:
This was actually done to be able to use MavenDependency, BomDependency and BomDependencyNoVersion in the different comparators and datasets. It does introduce new types but considering this a tool - I chose readability, and less type conversions over new types.